app/domain/authentication/authn_k8s/initialize.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+    # Secrets used when persisting a K8s authenticator
+    class K8sAuthenticatorData
+      include ActiveModel::Validations
+      include AuthData
+      attr_reader :service_account_token, :ca_certificate, :k8s_api_url, :json_data
+
+      def initialize(json_data)
+        @json_data = json_data
+        return if json_data.empty?
+        
+        @service_account_token = auth_parameters['service-account-token']
+        @ca_certificate = auth_parameters['ca-cert']
+        @k8s_api_url = auth_parameters['api-url']
+      end
+
+      def json_parameter_names
+        [ 'service-account-token', 'ca-cert', 'api-url' ]
+      end
+
+      def json_present?
+        !auth_parameters.keys.empty?
+      end
+
+      validates(
+        :json_data,
+        json: true,
+        if: :json_present?
+      )
+
+      validates(
+        :service_account_token,
+        presence: true,
+        if: :json_present?
+      )
+
+      # Regex below is based off of the RFC which formally defines PEM format, we only accept PEM
+      # not DER because binary data is difficult to handle well in JSON.
+      # https://www.rfc-editor.org/rfc/rfc7468
+      validates(
+        :ca_certificate,
+        format: {
+          with: %r{(((-+)BEGIN [A-Z ]+(-+)\n)(?:[A-Za-z\d+/]{4})*\n((-+)END [A-Z ]+(-+))\n?)},
+          message: "Certificate must be in PEM format"
+        },
+        presence: true,
+        if: :json_present?
+      )
+
+      validates(
+        :k8s_api_url,
+        url: true,
+        presence: true,
+        if: :json_present?
+      )
+    end
+
+    # Performs setup for newly persisted k8s authenticators
+    class InitializeK8sAuth
+      extend CommandClass::Include
+      include AuthorizeResource
+      
+      command_class(
+        dependencies: {
+          conjur_ca_repo: Repos::ConjurCA,
+          secret: Secret,
+          logger: Rails.logger
+        },
+        inputs: %i[conjur_account service_id auth_data current_user]
+      ) do
+        def call
+          loaded_secrets = @auth_data.auth_parameters&.each do |key, value|
```

Comment:
`each` doesn't return anything.  It just operates on the object (and in this case, the variable `loaded_secrets` will be the result of the last `@secret.create(resource_id: variable_id, value: value)` call).